### PR TITLE
s3 storage implementation/ index loading/ task & job status management/

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
- s3 storage implementation
- index loading
- task & job status management
## Modules
- [ ] UI
- [X] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [X] Others

## Checklist
- [x] run code format and lint check
- [ ] add unit test
- [x] add necessary doc

### CODE REVIEW

File: server/controller-sdk/src/main/java/ai/starwhale/mlops/domain/swds/index/SWDSBlock.java [hunk 1]
```diff
@@ -7,11 +7,26 @@
...
 package ai.starwhale.mlops.domain.swds.index;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import lombok.Builder;
 import lombok.Data;
 
 /**
  * the storage unit of one data set
+ * {
+ *    "id": 1,          //第n条记录
```
Comment (tianweidut) [inline server/controller-sdk/src/main/java/ai/starwhale/mlops/domain/swds/index/SWDSBlock.java:17]: use English comment？

**Review by dreamlandliu — DISMISSED** [review]
LGTM

File: server/controller/src/main/java/ai/starwhale/mlops/domain/task/LivingTaskStatusMachineImpl.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    @Override
+    public void adopt(Collection<Task> livingTasks, final TaskStatus status) {
+        //watch status change validation
+        //update to taskStatusMap & taskIdMap
+        //if in transaction context save immediately or add to toBePersistent
+        boolean inTransactionContext = false;
+        try {
+            TransactionAspectSupport
+                .currentTransactionStatus();
```
Comment (tianweidut) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/task/LivingTaskStatusMachineImpl.java:81]: ```TransactionAspectSupport.currentTransactionStatus();```
in one line?

**Review by tianweidut — CHANGES_REQUESTED** [review]

File: server/controller/src/main/java/ai/starwhale/mlops/domain/task/LivingTaskStatusMachineImpl.java (hunk 2 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/task/LivingTaskStatusMachineImpl.java]: done

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**